File: back_end/RB101-109/interview_assessment/small_problems/easy_6/2.rb
=begin

Write a method that takes an array of strings, and returns an array of the same string values, except with the vowels (a, e, i, o, u) removed.

input: array of strings
output: array of string without vowels
rules: 
- vowels are aeiou
- return new array
- preserve letter case
data structure: array, strings
pseudo-code: 
- iterate over the array of strings
- delete all vowels from the words
- return a new array of word values
algorithm: 
- map over the array of strings and delete each vowel
- capture vowel values: "aeiouAEIOU"
- return new array 
=end

def remove_vowels(array)
  array.map { |word| word.delete("aeiouAEIOU") }
end

p remove_vowels(%w(abcdefghijklmnopqrstuvwxyz))
p remove_vowels(%w(green YELLOW black white))
p remove_vowels(%w(ABC AEIOU XYZ)) == ['BC', '', 'XYZ']

File: exercises/js_small_problems/string_text_processing/2.js
// Write a function that takes an array of strings and returns an array of the same string values, but with all vowels (a, e, i, o, u) removed.
//

/*
input: array of strings of letters
output: new array of strings of letters with vowels removed
rules:
- vowels are a, e, i , o and u
algorithm:
- iterate over the array of words:
  - for each word, replace the vowels with empty string
- return the new array
*/

function removeVowels(array) {
  return array.map(word => word.replace(/[aeiou]/gi, ''));
}

console.log(removeVowels(['abcdefghijklmnopqrstuvwxyz']));         // ["bcdfghjklmnpqrstvwxyz"]
console.log(removeVowels(['green', 'YELLOW', 'black', 'white']));  // ["grn", "YLLW", "blck", "wht"]
console.log(removeVowels(['ABC', 'AEIOU', 'XYZ']));                // ["BC", "", "XYZ"]
